src/discord/rest_api/discord_endpoints.rs

```diff
@@ -117,6 +118,22 @@ trait ApiRes: Sized + for<'a> Deserialize<'a> {
     }
 }
 
+//Temporary function, we will remove it once we have a proper way to make non-blocking API requests.
+pub(crate) fn get_discord_user_info() -> AuthedUser {
+    let (tx, rx) = oneshot::channel();
+    runtime().spawn(async move {
+        let mut headers = HashMap::new();
+        headers.insert("Authorization", get_tokens().unwrap().discord_token.unwrap());
+
+        let messages = ApiEndpoints::GetUser.call(headers).await.unwrap();
+        tx.send(messages).unwrap();
+    });
+    if let ApiResponse::User(user) = rx.blocking_recv().unwrap() {
+        user
+    } else {
+        panic!("User data not found.")
+    }
+}
```

Comment:
This is not a direct call to the discord server this is a util function that should go to the discord/util.rs